src/java.base/share/classes/java/util/zip/ZipCoder.java

```diff
@@ -273,7 +280,7 @@ int checkedHash(byte[] a, int off, int len) throws Exception {
                 // shared and that decoder is not thread safe.
                 // We use the JLA.newStringUTF8NoRepl variant to throw
                 // exceptions eagerly when opening ZipFiles
-                return hash(JLA.newStringUTF8NoRepl(a, off, len));
+                return hash(toString(a, off, len));
```

Comment:
Earlier, `byte[]` was shared by the `String` instantiated, now it needs to be cloned. That is, I presume this to have a performance implication. Was this considered?

Reply:
It always makes a copy, previously and as refactored.
The trampoline in System.java called into String with noShare = true.
```
public String newStringUTF8NoRepl(byte[] bytes, int off, int len) {
    return String.newStringUTF8NoRepl(bytes, off, len, true);
}
```
Now the copy is explicitly done by the caller and the contract is clear that the buffer is always not shared but transferred. (and one less argument and conditional)